client/http-client.js

```diff
@@ -1,14 +1,16 @@
 const qs = require('qs')
 const fetch = require('node-fetch')
+const HttpsProxyAgent = require('https-proxy-agent')
 
 module.exports = (clientConfig) => {
   // setup proxy agent in case a proxy is configured
   let agent = null
   if (clientConfig.agent) {
     agent = clientConfig.agent
   } else if (clientConfig.proxy) {
-    const HttpsProxyAgent = require('https-proxy-agent')
     agent = new HttpsProxyAgent(clientConfig.proxy)
+  } else if (process.env.HTTP_PROXY) {
```

Comment:
I'd still like to have the possibility to disable the proxy, even if `HTTP_PROXY` is present.
So if `clientConfig.proxy === false` then we also want to ignore the `HTTP_PROXY` option.

This is useful if there are other services in the process that need a proxy.
@disufi could you also check how the `https-proxy-agent` module behaves with a `NO_PROXY` variable?

Reply:
@marcbachmann it simply ignores all env variables, no behaviour at all..

Reply:
ok, so we definitely need the `proxy: false` support to override `HTTP_PROXY`

Reply:
usually `NO_PROXY` is there to prevent the usage of a proxy for specific host or ip address ranges, we're using that in the bluewin setup and in our test suite

Reply:
I didn't know you have such use cases. If the "environment" provides configuration, I expect it should be respected, otherwise, well it shouldn't be provided.

I added this, because of the simple support usage of `https-proxy-agent`. This was additional "basic support" (a more complete one would probably look like this: https://github.com/gajus/xfetch/blob/master/src/index.js#L125). Anything more complex (disabling and configuring) can anyway be done with the option to provide custom agent.

I feel that the addition of `proxy: false`, to override `HTTP_PROXY` is a bit too complex, so I would rather close the PR and continue using it as is.

Reply:
Yes, no problem at all. That's the reason I jumped in... because I knew that there might be problems. The main issue is that other services might be hosted in a different network and that results in a conflict with the config.

Mainly we just have to decide how we want to handle it.
We currently use sdk at swisscom and they have a set up where files on a s3-like api are remote and the livingdocs service is local.
As reference for others who want to check the code, here is it: https://github.com/livingdocsIO/livingdocs-swisscom-tv-delivery/blob/9ee3720ab8102bc93f8cb0c0d47caeb4229e39ad/lib/livingdocs/routes.js#L9

cc @disufi